### PR TITLE
Add little helper function for printing pandora docs

### LINKED ISSUE #52: Make function to call `TAB_Field_Comment`
This contains all the help info... might be a good reference

```
> sidora_entry_help("TAB_Capture", "Files")
Files such as quantification results (if performed).

```

### PR DESCRIPTION
**Pull request by jfy133**
WIP because I can't seem to add `TAB_Field_Comment` to the pandora_tables_all, but if I manually load with `get_con` it works.

Also added crayon for pretty printing as an experiement. If no problems we could extend to all help/warn messages for :eye: :candy: 

Fixes #52

### CODE REVIEW

**Comment by jfy133** [comment]
@nevrome I've tried running all the commands in `data-raw/data_prep.R` but can't get TAB_Field_Comment to propogate to `pandora_tables_all.rda` :(

File: R/utils_general.R [hunk 1]
```diff
@@ -23,3 +23,44 @@ check_if_packages_are_available <- function(packages_ch) {
...
+  if ( !tab %in% sidora.core::pandora_tables ) {
+    stop(crayon::red("[sidora.core] error: "), "Select one valid PANDORA SQL table.")
+    
+  }
+  
+  if (!field %in% sidora.core::pandora_column_types$col_name ) {
+    stop(crayon::red("[sidora.core] error: "), "Field name is not recognised.")
+  }
+  
+  pandora_help <- sidora.core::get_df("TAB_Field_Comment", con) %>% tibble::as_tibble()
```
Comment (nevrome) [inline R/utils_general.R:53]: Why the `%>% tibble::as_tibble()`? Doesn't  `get_df` already yield a tibble? :thinking:

File: R/utils_general.R [hunk 2]
```diff
@@ -23,3 +23,44 @@ check_if_packages_are_available <- function(packages_ch) {
...
+  result <- dplyr::filter(pandora_help, .data$Table == tab, .data$Field == field) %>% 
+    dplyr::pull(.data$Comment)
```
Comment (nevrome) [inline R/utils_general.R:55]: What happens if there are two identical entries in `TAB_Field_Comment`?

File: R/utils_general.R [hunk 3]
```diff
@@ -23,3 +23,44 @@ check_if_packages_are_available <- function(packages_ch) {
     )
   }
 }
+
+#' Access field Pandora help tooltip
+#'
+#' This functions allows you to call the 'tooltip' help/documentation message
+#' present on the Pandora webpage for a given entry.
+#'
+#' @param tab character. Name of the table of the field of interest
```
Comment (nevrome) [inline R/utils_general.R:32]: In which format is this name? `TAB_*`?

**Comment by nevrome** [comment]
`TAB_Field_Comment` is not correctly loaded from `pandora_column_types.tsv`, because you forgot to add two columns.

```r
> pandora_column_types <- readr::read_tsv(
+   "data-raw/pandora_column_types.tsv",
+   col_types = readr::cols(
+     table = readr::col_character(),
+     entity_type = readr::col_character(),
+     col_name = readr::col_character(),
+     type = readr::col_character(),
+     auxiliary_table = readr::col_character(),
+     update_col_type = readr::col_character()
+   )
+ )
Warning message:                                                                                                                         
One or more parsing issues, see `problems()` for details 
> readr::problems()
# A tibble: 3 × 5
    row   col expected  actual    file                                                            
  <int> <int> <chr>     <chr>     <chr>                                                           
1    48     5 7 columns 5 columns /home/schmid/agora/sidora.core/data-raw/pandora_column_types.tsv
2    49     5 7 columns 5 columns /home/schmid/agora/sidora.core/data-raw/pandora_column_types.tsv
3    50     4 7 columns 4 columns /home/schmid/agora/sidora.core/data-raw/pandora_column_types.tsv
```

**Comment by nevrome** [comment]
I'm skeptical if it is a good idea to add `crayon`. Just adds more maintenance work and an additional dependency for a pretty questionable, aesthetic goal... And it's not really pleasant if only one function supports it.

File: R/utils_general.R (hunk 1 above)
Comment (jfy133) [reply R/utils_general.R]: Left over from when I was running `get_con`

**Comment by jfy133** [comment]
> I'm skeptical if it is a good idea to add crayon. Just adds more maintenance work and an additional dependency for a pretty questionable, aesthetic goal... And it's not really pleasant if only one function supports it.

Ok, fine you can continue living in your grey, concrete industrial world... removed ;)

File: README.md [hunk 4]
```diff
@@ -216,6 +216,6 @@ A special hint concerning `TAB_Analysis`: It is formatted differently from the o
 
 Additionally, if you've made a modification it's recommended to run build-validation before rebuilding. This can typically be accessed in Rstudio with  <kbd>Ctrl</kbd> + <kbd>shift</kbd> + <kbd>e</kbd>
 
-If you need to update any field names or add new tables, make sure to update the TSV in `data-raw`, and run the `data_prep.R` script. Don't be stupid like James and edit the TSVs in the Rstudio text editor - do it in a spreadsheet tool. 
+If you need to update any field names or add new tables for Pandora, make sure to update the TSV in `data-raw`, and run the `data_prep.R` script. The .tsv files can be edited in a spreadsheet tool like [LibreOffice](https://www.libreoffice.org).
```
Comment (jfy133) [inline README.md:219]: Ugh fine be professional too ;)

**Comment by nevrome** [comment]
What I only realized now: `field_help()` only prints the help to the console. This is a fun thing for interactive date exploration, but not terribly useful for automatic scripts. I suggest that the function should not pretty print anything and instead just return a string with the help message. The function should probably also be renamed to `get_field_help()` to better fit in with our other information download functions.

It looks like I'm shutting down anything fun :sweat_smile:. But I just want to keep the interface clean and useful.

File: README.md (hunk 4 above)
Comment (nevrome) [reply README.md:219]: Haha - no self-deprecating humor, please :stuck_out_tongue:

**Comment by jfy133** [comment]
> What I only realized now: `field_help()` only prints the help to the console. This is a fun thing for interactive date exploration, but not terribly useful for automatic scripts. I suggest that the function should not pretty print anything and instead just return a string with the help message. The function should probably also be renamed to `get_field_help()` to better fit in with our other information download functions.
> 
> It looks like I'm shutting down anything fun sweat_smile. But I just want to keep the interface clean and useful.

1. I'm not sure what you mean here - I can't think of any use case when you would ever want to get the help 'tooltip' information in an automatic script. The purpose of this function is exactly for quick reference during interactive sessions, unless you can think of possible usecases? For readability purposes I would rather keep it with `cat` 
2. Agree to rename as `get_`!

**Comment by nevrome** [comment]
One possible usecase might be report generation. You could theoretically use `get_field_help()` to add mouse-hover help tooltips for summary tables. Any time you want to prepare an alternative interface to explore Pandora tables this could become useful. And your interactive usecase is covered as well with a clean string.

**Comment by jfy133** [comment]
> One possible usecase might be report generation. You could theoretically use `get_field_help()` to add mouse-hover help tooltips for summary tables. Any time you want to prepare an alternative interface to explore Pandora tables this could become useful. And your interactive usecase is covered as well with a clean string.

Not entirely convinced this would ever be done (the help messages probably need updating/and are often ugly formatted), but sure I guess theoretically could happen.

**Review by nevrome — APPROVED** [review]
Nice :+1:

**Pull request merged**